bin/build_libmonero_cpp.sh

```diff
@@ -1,9 +1,51 @@
 #!/bin/sh
 
+export HOST_NCORES=${HOST_NCORES-$(nproc 2>/dev/null || shell nproc 2>/dev/null || sysctl -n hw.ncpu 2>/dev/null || echo 1)}
+BUILD_DIR="$(pwd)/build"
+INSTALL_DIR="$BUILD_DIR/install"
+export CMAKE_PREFIX_PATH=$INSTALL_DIR${CMAKE_PREFIX_PATH+:$CMAKE_PREFIX_PATH}
+USE_DEVICE_TREZOR=${USE_DEVICE_TREZOR-ON}
+echo "HOST_NCORES=$HOST_NCORES in $0"
+echo "CMAKE_PREFIX_PATH=$CMAKE_PREFIX_PATH in $0"
+echo "USE_DEVICE_TREZOR=$USE_DEVICE_TREZOR in $0"
+
+[ -d $INSTALL_DIR ] || mkdir -p $INSTALL_DIR
+
+(cd external/boost && \
```

Comment:
I just need to ensure these additions don't interfere with the build process in monero-ts.